File: ShadowFrameworkLite20/src/shadow/animation/data/SFCompositeAnimationData.java
package shadow.animation.data;

import shadow.animation.SFAnimation;
import shadow.animation.SFCompositeAnimation;
import shadow.renderer.data.SFDataAsset;
import shadow.renderer.data.SFDataAssetList;
import shadow.system.data.SFNamedParametersObject;

public class SFCompositeAnimationData extends SFDataAsset<SFAnimation>{

	private SFDataAssetList<SFAnimation> animations;
	
	public SFCompositeAnimationData() {
		animations=new SFDataAssetList<SFAnimation>();
		SFNamedParametersObject namedParameters=new SFNamedParametersObject();
		namedParameters.addObject("animations", animations);
		setData(namedParameters);
	}
	
	public void addAnimation(SFDataAsset<SFAnimation> animation){
		animations.add(animation);
	}
	
	@Override
	protected SFCompositeAnimation buildResource() {
		SFCompositeAnimation animation=new SFCompositeAnimation();
		
		for (int i = 0; i < animations.size(); i++) {
			animation.add(animations.get(i).getResource());
		}
		
		return animation;
	}
}

File: ShadowFrameworkJS/src/shadow/animation/data/SFCompositeAnimationData.js



function SFCompositeAnimationData(){
		this.animations=new SFDataAssetList();
		var namedParameters=new SFNamedParametersObject();
		namedParameters.addObject( this.animations);
		this.setData(namedParameters);	
}

inherit(SFCompositeAnimationData,SFDataAsset);

SFCompositeAnimationData.prototype["buildResource"]=function(){
	var animation=new SFCompositeAnimation();
		
	for (var i = 0; i < this.animations.length; i++) {
		animation.add(this.animations[i].getResource());
	}
		
	return animation;
};

SFCompositeAnimationData.prototype["generateNewDatasetInstance"]=function(){
			return new SFCompositeAnimationData();
		};
		